tool/gravity/cli/checks.go

```diff
@@ -17,37 +17,89 @@ limitations under the License.
 package cli
 
 import (
+	"context"
 	"fmt"
 	"io/ioutil"
+	"time"
 
+	"github.com/gravitational/gravity/lib/app"
 	"github.com/gravitational/gravity/lib/checks"
+	"github.com/gravitational/gravity/lib/constants"
+	"github.com/gravitational/gravity/lib/fsm"
 	"github.com/gravitational/gravity/lib/localenv"
+	"github.com/gravitational/gravity/lib/pack"
 	"github.com/gravitational/gravity/lib/schema"
+	statusapi "github.com/gravitational/gravity/lib/status"
+	upgradechecks "github.com/gravitational/gravity/lib/update/cluster/checks"
 
+	"github.com/fatih/color"
 	pb "github.com/gravitational/satellite/agent/proto/agentpb"
 	"github.com/gravitational/trace"
 )
 
-func checkManifest(env *localenv.LocalEnvironment, manifestPath, profileName string, autoFix bool) error {
-	data, err := ioutil.ReadFile(manifestPath)
+type preflightChecksConfig struct {
+	manifestPath string
+	imagePath    string
+	profileName  string
+	autoFix      bool
+	timeout      time.Duration
+}
+
+func executePreflightChecks(env *localenv.LocalEnvironment, config preflightChecksConfig) error {
+	ctx, cancel := context.WithTimeout(context.Background(), config.timeout)
+	defer cancel()
+
+	if !detectCluster(ctx) {
+		env.PrintStep("No deployed cluster detected, running install preflight checks")
+		return checkInstall(ctx, env, config)
+	}
+
+	env.PrintStep("Detected deployed cluster, running upgrade preflight checks")
+	return checkUpgrade(ctx, env, config)
+}
+
+// detectsCluster attempts to detect whether the current node has a cluster deployed.
+//
+// Returns true if the cluster is detected and false otherwise.
+func detectCluster(ctx context.Context) bool {
+	_, err := statusapi.FromPlanetAgent(ctx, nil)
 	if err != nil {
-		return trace.Wrap(err)
+		log.WithError(err).Info("No cluster detected: failed to query planet agent.")
```

Comment:
Hmm, I think logging this could cause some confusion. It's tough, because this particular log is completely normal on a new install, and abnormal on an upgrade.

Reply:
Yeah, that's why I logged it as info. The error is there to ease possible troubleshooting in cases we discussed below (and above).